rancher/distros-test-framework
# [Not Ready] Add dual stack tests

#### Proposed Changes ####


- This PR adds the ability to run dual stack tests (Cluster with both IPV4 and IPV6 IPs) which includes validation with ingress, nodeport, clusterIP and multi-stack deployments.
- The test is configured to run with a Bastion instance created on the same subnet as the server and agent nodes as traditional way of test execution were causing failures when trying to validate IPV6 IPs
- The test needs to be executed on AWS us-west-1 region as the security group, vpc and subnet ids specific to dual stack setting is already configured in that region
- Additional variables are added keeping in mind for airgap and ipv6 testing, which will be added in upcoming PRs


- Yes, the changes require updating documentation

#### Types of Changes ####


- Adding dual stack tests
- Updating existing vars: 
  - In k3s.tfvars: `instance_class` -> `db_instance_class` and `environment` -> `db_environment`
- Additional Variables in *.tfvars:
```
no_of_bastion_nodes = 1
bastion_subnets     = "<subnet-of-bastion-instance>"
enable_public_ip   = true
enable_ipv6        = true
```
- Required Variable: 
```
server_flags = "cluster-cidr: <IPV4-CIDR-1>,<IPV6-CIDR-1>\nservice-cidr: <IPV4-CIDR-2>,<IPV6-CIDR-2>\n"
```

#### Testing ####


Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.
- Ran the PR on both k3s and rke2 Jenkins job and shared the results

2. Verify code lint; we should not have errors.
- Locally having some issue so depending in the CI linting

3. Update the documentation if needed.
- TBA

4. Update makefile and docker run if adding new tests.
- Updated makefile

5. Run your tests at least 4 times with all configurations needed and possible.
- The code continues to work with existing settings

6. If needed test with different os types.
- Only ran with Ubuntu in us-west-1, if needed will execute on other OS types

#### Linked Issues ####


Ref Ticket: https://github.com/rancher/qa-tasks/issues/882

#### Further Comments ####
